GTG/core/timer.py

```diff
@@ -70,15 +72,59 @@ def time_changed(self):
         self.timeout_source = GObject.timeout_add_seconds(secs_to_refresh,
                                                           self.emit_refresh)
 
-    def set_configuration(self, refresh_hour, refresh_min):
+    def set_configuration(self, time):
+        self.config.set('hour', time.hour)
+        self.config.set('min', time.minute)
+
+    def get_configuration(self):
         try:
-            datetime.time(int(refresh_hour), int(refresh_min))
-        except (ValueError, TypeError):
-            raise ValueError("{}:{} is not a valid time".format(refresh_hour,
-                                                                refresh_min))
+            return datetime.time(int(self.config.get('hour')),
+                                 int(self.config.get('min')))
+        except(ValueError):
+            Log.error("Invalid time values. Try using HH:MM or am/pm format")
+            return datetime.time(0, 0)
 
-        self.config.set('hour', refresh_hour)
-        self.config.set('min', refresh_min)
+    def parse_time(self, time):
+        """
+        This function parses user input when setting
+        start of the day in general preferences.
+        input:  @time               user's input as string
 
-    def get_configuration(self):
-        return self.config.get('hour'), self.config.get('min')
+        output: @hour               default format for hours
+                @minute             default format for minutes
+                @ValueError         raised if @time is not valid
+        """
+        time = time.lower().strip()
+        invalid_format = False
+        # Format for HH:MM and optional am/pm
+        match = re.match(r'^(?P<hour>[0-9]{1,2}):?(?P<minute>[0-9]{2})?'
+                         r' ?(?P<am_pm>[ap]m)?$', time)
+        if match:
+            hour = int(match.group('hour'))
+            minute = int(match.group('minute') or 0)
+            am_pm = match.group('am_pm')
+
+            if am_pm == 'am' and int(hour) == 12:
+                    hour = 0
+            elif am_pm == 'pm':
+                if 0 < int(hour) < 12:
+                    hour = hour + 12
+                elif hour == 12:
+                    hour = 12
+                elif hour > 12:
+                    invalid_format = True
+
+            if not invalid_format:
+                return datetime.time(int(hour), int(minute))
+            else:
+                raise ValueError('This time value or format\
+                                 is not allowed: {0}'.format(time))
+        else:
+            # The final attempt to parse the input
+            return datetime.datetime.strptime(time, '%X')
+
+    def get_formatted_time(self):
+        formated_time = self.get_configuration().strftime("%X")
+        correct_locale_time = re.sub(r'([0-9]{1,2}:[0-9]{2}):[0-9]{2}', r'\1',
```

Comment:
You might want to either return it directly or keep it `formatted_time`. `correct_locale_time` sounds strange to me (and it might not be correct 100% time)